input/system/azure/system.go

```diff
@@ -0,0 +1,320 @@
+			logger.PrintError("Azure/System: Failed to make a Cosmos DB client: %v\n", err)
+			return
+		}
+		// Search a server from the list
+		pager := clientFactory.NewClustersClient().NewListPager(nil)
+		for pager.More() {
+			page, err := pager.NextPage(ctx)
+			if err != nil {
+				logger.PrintError("Azure/System: Failed to advance page of Cosmos DB cluster list: %v\n", err)
+				break
+			}
+			for _, v := range page.Value {
+				if v.ID != nil {
+					resourceIDParts := strings.Split(*v.ID, "/")
+					resourceGroup := resourceIDParts[4]
+					resourceType := resourceIDParts[7]
+					resourceName := resourceIDParts[8]
+
+					if config.AzureDbServerName == resourceName {
+						customWindowEnabled := util.StringCustomTypePtrToString(v.Properties.MaintenanceWindow.CustomWindow) == "Enabled"
+						system.Info.Azure = &state.SystemInfoAzure{
+							Location:                 util.StringPtrToString(v.Location),
+							CreatedAt:                util.TimePtrToTime(v.SystemData.CreatedAt),
+							State:                    util.StringCustomTypePtrToString(v.Properties.State),
+							SubscriptionID:           config.AzureSubscriptionID,
+							ResourceGroup:            resourceGroup,
+							ResourceType:             resourceType,
+							ResourceName:             resourceName,
+							MaintenanceCustomWindow:  customWindowEnabled,
+							MaintenanceDayOfWeek:     util.Int32PtrToInt(v.Properties.MaintenanceWindow.DayOfWeek),
+							MaintenanceStartHour:     util.Int32PtrToInt(v.Properties.MaintenanceWindow.StartHour),
+							MaintenanceStartMinute:   util.Int32PtrToInt(v.Properties.MaintenanceWindow.StartMinute),
+							CitusVersion:             util.StringPtrToString(v.Properties.CitusVersion),
+							HighAvailabilityEnabled:  util.BoolPtrToBool(v.Properties.EnableHa),
+							CoordinatorStorageMB:     util.Int32PtrToInt(v.Properties.CoordinatorStorageQuotaInMb),
+							NodeStorageMB:            util.Int32PtrToInt(v.Properties.NodeStorageQuotaInMb),
+							CoordinatorVCores:        util.Int32PtrToInt(v.Properties.CoordinatorVCores),
+							NodeVCores:               util.Int32PtrToInt(v.Properties.NodeVCores),
+							CoordinatorServerEdition: util.StringPtrToString(v.Properties.CoordinatorServerEdition),
+							NodeServerEdition:        util.StringPtrToString(v.Properties.NodeServerEdition),
+							NodeCount:                util.Int32PtrToInt(v.Properties.NodeCount),
+						}
+						tags := make(map[string]string)
+						for key, value := range v.Tags {
+							tags[key] = util.StringPtrToString(value)
+						}
+						system.Info.ResourceTags = tags
+						resourceID = *v.ID
+						break
+					}
+				}
+			}
+		}
+	}
+
+	if resourceID == "" {
+		server.SelfTest.MarkCollectionAspectWarning(state.CollectionAspectSystemStats, "unable to find the database server info")
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the Monitoring Reader permission of the database is granted to the managed identity.")
```

Comment:
Does this need to be moved to the earlier parts where we actually make the list API calls? (since we return early on error) -- or do we just not get any results if the permissions are insufficient?

Reply:
Just to close the loop: I added the comment above, but this can be reached when the managed identity is assigned to some db but not this db, so I think the wording is still accurate. When it's not assigned to any db resources, it'll raise an error when we do `page, err := pager.NextPage(ctx)` earlier in line 46.

I tweaked that part of code, to make sure that it'll early return when it receives errors (and show the same hint). It'll look like the following. It's not super good looking sadly (I think %v is just showing too much), but I think it does show good info for debugging so I'd say it's a good start.

![image](https://github.com/user-attachments/assets/ee7795df-8c37-4ebc-82ee-52d62620bcd1)